numba/core/typing/npydecl.py

```diff
@@ -495,6 +495,14 @@ class NpArray(CallableTemplate):
     def generic(self):
         def typer(object, dtype=None):
             ndim, seq_dtype = _parse_nested_sequence(self.context, object)
+            if isinstance(dtype.dtype, types.Record):
+                # a nested sequence should be interpreted as ndim=1 array,
+                # with the other dimension being part of the Record
+                if ndim == 2:
+                    ndim = 1
```

Comment:
Only if the nested sequence is of type tuple, I think.

Reply:
a list of lists is allowed, while a tuple of lists is not. This is line with Numpy behaviour. I've added tests to cover both combinations that should work: list of tuples and list of lists.
I also added a test to cover two wrong inputs: flat list and 3-nested list.

Reply:
I was under the impression that numpy only allowed a list of tuples.